src/toolkitAPI.c

```diff
@@ -1090,6 +1090,59 @@ int DLLEXPORT swmm_getSystemRunoffStats(TRunoffTotals *runoffTot)
     return(errorcode);
 }
 
+int DLLEXPORT swmm_getGageParam(int index, int Param, double *value)
+//
+// Input:   index = Index of desired ID
+//          param = Parameter desired (Perhaps define enum)
+// Output:  value = value to be output
+// Return:  API Error
+// Purpose: Gets Gage Parameter
+{
+    // Check if Open
+    if(swmm_IsOpenFlag() == FALSE) return(ERR_API_INPUTNOTOPEN);
+        // Check if object index is within bounds
+    if (index < 0 || index >= Nobjects[GAGE]) return(ERR_API_OBJECT_INDEX);
+
+    switch(Param)
+    {
+        // raintype
+        case 0: *value = Gage[index].rainType; break;
+        // rainfall value
+        case 1: *value = Gage[index].rainfall / UCF(RAINFALL); break;
```

Comment:
How do we deal with an external file or injection from external script/program where we don't know if the units are in the same units as the rest of the input file? Should we somehow protect the user from converting something that doesn't need to be converted or is there no clear way around this?